daemon/stop.go

```diff
@@ -23,6 +23,9 @@ func (daemon *Daemon) ContainerStop(name string, seconds *int) error {
 	}
 	if !container.IsRunning() {
 		err := fmt.Errorf("Container %s is already stopped", name)
+		if container.IsCreated() {
+			return errNotRunning{container.ID}
```

Comment:
Although codes here return an error, I think we still need to pay more attention on the status code docker daemon generated here.
While codes here you added seems to produce a status code of 500 which seems needing more discussion.